python/pyspark/rdd.py

```diff
@@ -877,6 +877,19 @@ def collect(self):
             sock_info = self.ctx._jvm.PythonRDD.collectAndServe(self._jrdd.rdd())
         return list(_load_from_socket(sock_info, self._jrdd_deserializer))
 
+    def collectWithJobGroup(self, groupId, description, interruptOnCancel=False):
+        """
+        .. note:: Experimental
+
+        When collect rdd, use this method to specify job group.
+
+        .. versionadded:: 3.0.0
```

Comment:
Do we target this at 3.0.0?
cc @gatorsmile

Reply:
It would be great if we have @HyukjinKwon [explanation](https://github.com/apache/spark/pull/28395#pullrequestreview-402300959) on the PR description.

Reply:
BTW, we cannot remove this API forever. cc @marmbrus 
> I will target to deprecate this API in Spark 3.1.

Reply:
I dont think we will ignore the existing definitions of API stability tags. Might be best to stick to what's documented there ...